dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx

```diff
@@ -0,0 +1,1701 @@
+                          type="radio"
+                          name="auth"
+                          value={
+                            PackageRepositoryAuth_PackageRepositoryAuthType[
+                              PackageRepositoryAuth_PackageRepositoryAuthType
+                                .PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
+                            ]
+                          }
+                          checked={
+                            helmPSAuthMethod ===
+                            PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
+                          }
+                          onChange={handleImgPSChange}
+                          disabled={
+                            !!(repo?.customDetail as Partial<RepositoryCustomDetails>)
+                              ?.dockerRegistrySecrets?.length
+                          }
+                          required={
+                            authMethod ===
+                            PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
+                          }
+                        />
+                      </CdsRadio>
+                    </CdsRadioGroup>
+                    {/* End imagePullSecrets selection */}
+                    {/* Begin imagePullSecrets details */}
+                    <div cds-layout="col@xs:8">
+                      {/* Begin docker creds authentication */}
+                      <div
+                        id="kubeapps-repo-imagePullSecrets-details-docker"
+                        hidden={
+                          helmPSAuthMethod !==
+                          PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
+                        }
+                      >
+                        {/* TODO(agamez): enable the selection once the API supports it */}
+                        {/* <CdsToggleGroup className="flex-v-center">
+                          <CdsToggle>
+                            <label>
+                              {isUserManagedPSSecret
+                                ? "Use user-managed secrets"
+                                : "Use Kubeapps-managed secrets"}
+                            </label>
+                            <input
+                              type="checkbox"
+                              onChange={handleIsUserManagedPSSecretChange}
+                              checked={isUserManagedPSSecret}
+                              disabled={
+                                !!(repo?.customDetail as Partial<RepositoryCustomDetails>)
+                                  ?.dockerRegistrySecrets?.length
+                              }
+                            />
+                          </CdsToggle>
+                        </CdsToggleGroup>
+                        <br /> */}
+                        {isUserManagedPSSecret ? (
```

Comment:
This involves having a separate toggle for the user to choose? Maybe we are adding to the cognitive load of users (two choices instead of one).
Why not checking the general `isUserManagedSecret` toggle/flag?

Reply:
I've implemented this way as I wasn't really sure of the existence of repos with, let's say: basic auth for accessing the index.yaml + image pull secrets for the images.
In this case, what if a user wants to set the values directly for the basic auth but they already have a Secret in their k8s cluster for the pull secrets? 

I mean, even if we'd be reducing the points of decisions, we would be removing flexibility, wouldn't we?

Reply:
> My rationale for the question was: if the user already selected "I will manage the secrets", then how likely is that he/she already created the secrets for the pullSecrets?

Yep, I agree, it's really unlikely wanting to select something different. What if, if the field hasn't been populated yet, set the pullSecret's toggle to the same value. This way we will be suggesting sth to the user, but not forcing to.

> Yes, they are. I feel more natural to write the actual values for the secrets than the secret names from k8s.
Also, note that the secret-selector (re)implementation is still pending, so the UX for this is a little bit worse.